MSALiOS/ViewController.swift

```diff
@@ -33,196 +33,232 @@ import MSAL
+                return cachedAccounts[0]
             }
-            
+
+        } catch let error as NSError {
+
+            self.updateLogging(text: "Didn't find any accounts in cache: \(error)")
         }
-        
-    } catch {
-        
-        // This is the catch all error.
-        
-        self.loggingText.text = "Unable to acquire token. Got error: \(error)"
-        
+
+        return nil
     }
+
+    func updateLogging(text : String) {
+
+        if Thread.isMainThread {
+            self.loggingText.text = text
+        } else {
+            DispatchQueue.main.async {
+                self.loggingText.text = text
+            }
+        }
     }
 
+    func updateSignoutButton(enabled : Bool) {
+        if Thread.isMainThread {
+            self.signoutButton.isEnabled = enabled
+        } else {
+            DispatchQueue.main.async {
+                self.signoutButton.isEnabled = enabled
+            }
+        }
+    }
     
     /**
-     This button will invoke the call to the Microsoft Graph API. It uses the
-     built in Swift libraries to create a connection.
-     Pay attention to the error case below. It shows you how to
-     detect a `UserInteractionRequired` Error case and present the `acquireToken()`
-     method again for the user to sign in. This usually happens if
-     the Refresh Token has expired or the user has changed their
-     password.
-     
+        This button will invoke the call to the Microsoft Graph API. It uses the
+        built in URLSession to create a connection.
      */
 
- func getContentWithToken() {
-    
-    let sessionConfig = URLSessionConfiguration.default
-    
-    // Specify the Graph API endpoint
-    let url = URL(string: kGraphURI)
-    var request = URLRequest(url: url!)
-    
-    // Set the Authorization header for the request. We use Bearer tokens, so we specify Bearer + the token we got from the result
-    request.setValue("Bearer \(self.accessToken)", forHTTPHeaderField: "Authorization")
-    let urlSession = URLSession(configuration: sessionConfig, delegate: self, delegateQueue: OperationQueue.main)
+    func getContentWithToken() {
+
+        // Specify the Graph API endpoint
+        let url = URL(string: kGraphURI)
+        var request = URLRequest(url: url!)
     
-    urlSession.dataTask(with: request) { data, response, error in
+        // Set the Authorization header for the request. We use Bearer tokens, so we specify Bearer + the token we got from the result
+        request.setValue("Bearer \(self.accessToken)", forHTTPHeaderField: "Authorization")
+
+        URLSession.shared.dataTask(with: request) { data, response, error in
         
         let result = try? JSONSerialization.jsonObject(with: data!, options: [])
-                    if result != nil {
-                
-                self.loggingText.text = result.debugDescription
+
+            if result != nil {
+                self.updateLogging(text: result.debugDescription)
```

Comment:
it will add "Optional(..." to the string, I would do `if let result = result` here.